### PR TITLE
app image builder feature

### PR DESCRIPTION
**Pull request by fang64**
I have created a PR to create AppImages using GitHub Actions Workflow. 

The main modification I made was add a flag to BuildLinux.sh so it'll skip the memory check as it's not a problem to compile with less ram. That ram check could be removed but I've just simply added a flag to skip it. 

I also added a check to see if you are attempting to run the appimage build in a docker container so it'll patch the appimagetool so it'll run and use the extract and run option in that scenario. GH Action workflow isn't in a container but when testing with [nektos/act](https://github.com/nektos/act) it was needed to confirm the build was working correctly. 

I've incorporated @zhaofengli 's fix  [Disable EGL for wxGLCanvas](https://github.com/zhaofengli/BambuStudio/tree/wx-no-egl) with my PR to get the appimage to work properly. 

I'm sending a similar PR to BambuStudio upstream, but this one does have an issue with the AppImage producing an error when running the application:
image

"Error running Javascript: file:///tmp/.mount_BambuSKzd8iH/resources/web/homepage/index.html?lang=en_GB:1:9: ReferenceError: Can't find variable: __wxOut0"

Not entirely sure what's triggering that as it's not happening to upstream Bambu Studio.

### CODE REVIEW

**Comment by SoftFever** [comment]
Fantastic! 
Let me give it a try :)

File: deps/wxWidgets/wxWidgets.cmake [hunk 1]
```diff
@@ -49,6 +49,7 @@ bambustudio_add_cmake_project(wxWidgets
         -DwxUSE_LIBJPEG=sys
         -DwxUSE_LIBTIFF=sys
         -DwxUSE_EXPAT=sys
+        -DwxUSE_GLCANVAS_EGL=OFF
```
Comment (SoftFever) [inline deps/wxWidgets/wxWidgets.cmake:52]: Can we use `if(CMAKE_SYSTEM_NAME STREQUAL "Linux")` here to make sure this change only works for Linux build?

Comment (fang64) [reply deps/wxWidgets/wxWidgets.cmake:52]: Alright I'll make the change.

Comment (fang64) [reply deps/wxWidgets/wxWidgets.cmake]: @SoftFever I've made changes to make this Linux only build logic. Hopefully that works I still don't know what's triggering that wxOut0 issue but apparently it happens just building it normally anyway.

**Review by SoftFever — APPROVED** [review]

**Comment by SoftFever** [comment]
Thanks, @fang64 .
I appreciate it.
I will merge this now.
I have no idea why that warning popup. We will treat that as a separate bug.

**Pull request merged**